src/main/java/org/sunbird/cb/hubservices/serviceimpl/UserUtilityService.java

```diff
@@ -0,0 +1,143 @@
+package org.sunbird.cb.hubservices.serviceimpl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import org.sunbird.cb.hubservices.cache.RedisCacheMgr;
+import org.sunbird.cb.hubservices.model.MultiSearch;
+import org.sunbird.cb.hubservices.model.Request;
+import org.sunbird.cb.hubservices.model.Search;
+import org.sunbird.cb.hubservices.profile.handler.ProfileUtils;
+import org.sunbird.cb.hubservices.service.IConnectionService;
+import org.sunbird.cb.hubservices.service.IUserUtility;
+import org.sunbird.cb.hubservices.util.ConnectionProperties;
+import org.sunbird.cb.hubservices.util.Constants;
+import org.sunbird.cb.hubservices.util.NetworkServerProperties;
+import org.sunbird.cb.hubservices.util.PrettyPrintingMap;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+@Service
+public class UserUtilityService implements IUserUtility {
+
+    @Autowired
+    RedisCacheMgr redisCacheMgr;
+
+    @Autowired
+    ConnectionProperties connectionProperties;
+
+    @Autowired
+    IConnectionService connectionService;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @Autowired
+    NetworkServerProperties networkServerProperties;
+
+    private Logger logger = LoggerFactory.getLogger(UserUtilityService.class);
+
+    @Override
+    public Map<String, Object> getUserInfoFromRedish(MultiSearch multiSearch, String[] sourceFields, List<String> connectionIdsToExclude) {
+        String departmentName = "";
+        List<String> includeFields = sourceFields != null && !Arrays.asList(sourceFields).isEmpty() ? Arrays.asList(sourceFields) : ProfileUtils.getUserDefaultFields();
+        List<String> tags = new ArrayList<>();
+        Map<String, Object> tagRes = new HashMap<>();
+        for (Search sRequest : multiSearch.getSearch()) {
+            departmentName = (String) sRequest.getValues().get(0);
+            departmentName = departmentName.trim().replaceAll(" ", "");
+            String userInformation = redisCacheMgr.getCache(Constants.USER_LIST + Constants.UNDER_SCORE + departmentName);
+            if (!ObjectUtils.isEmpty(userInformation)) {
```

Comment:
can we avoid negation ?
just switch the implementation or use isNotEmpty method if available...

Reply:
Done